### PR TITLE
chore: Derive `tls_codec` for `Node`.

### PR DESCRIPTION
**Pull request by duesee**
An easy one :-)

### CODE REVIEW

File: openmls/src/treesync/node.rs [hunk 1]
```diff
@@ -11,11 +12,27 @@ pub(crate) mod leaf_node;
...
+///         case leaf:   LeafNode leaf_node;
+///         case parent: ParentNode parent_node;
+///     };
+/// } Node;
+/// ```
+#[derive(
+    Debug, PartialEq, Eq, Clone, Serialize, Deserialize, TlsSize, TlsDeserialize, TlsSerialize,
+)]
 #[allow(missing_docs)]
 #[allow(clippy::large_enum_variant)]
```
Comment (kkohbrok) [inline openmls/src/treesync/node.rs:30]: My local clippy is not complaining if I remove those two allows. While you're touching this code, could you check if we can get rid of them?

**Review by kkohbrok — APPROVED** [review]
Looks good with an optional nit since it's somewhat unrelated.

File: openmls/src/treesync/node.rs (hunk 1 above)
Comment (duesee) [reply openmls/src/treesync/node.rs]: Removing code is always good :-)

**Comment by duesee** [comment]
Fixed-up and force-pushed to cleanup the `fixup!`. As this PR is easy enough I thought that force-pushing directly may spare use some time because it would invalidate your review later.

**Review by kkohbrok — APPROVED** [review]
Thanks!

**Pull request merged**